### ISSUE TITLE
support for MMDetection styled logging

### ISSUE THREAD

**Issue by fcakyon** [ultralytics/yolov5#3165]
## 🚀 Feature

Ability to log train/val loss&metrics in the same format with [MMDetection](https://github.com/open-mmlab/mmdetection) library

## Motivation

Famous [MMDetection](https://github.com/open-mmlab/mmdetection) library and awesome YOLOv5 libraries log the results in different format, it prevents us to compare YOLOv5 and MMDetection trainings on the same plot in any experiment tracking platform.

## Pitch

A new train argument such as `--mmdet_tags` can be introduced log train/val loss&metrics in the same format with [MMDetection](https://github.com/open-mmlab/mmdetection) library so that we can compare yolov5 amd mmdet models on the same plot

## Additional context

Refer to my pr: https://github.com/ultralytics/yolov5/pull/3115

**Comment by glenn-jocher** [comment]
@fcakyon thanks for the PR! If we are going to make a change we probably want to make a breaking change to the metric names and simply switch over to the MM names. 

What about detectron2 though? Do they have a 3rd naming convention seperate from MM and YOLOv5?

**Comment by fcakyon** [comment]
@glenn-jocher unfortunately detectron2 has a 3rd naming convention seperate from YOLOv5 and MMDetection. For instance it says `loss_box_reg` for bbox loss: https://github.com/facebookresearch/detectron2/blob/f50ec07cf220982e2c4861c5a9a17c4864ab5bfd/detectron2/modeling/roi_heads/fast_rcnn.py#L313

**Comment by glenn-jocher** [comment]
@fcakyon hmm what a mess.

@Borda does PyTorch Lightning align with either MMdetection or detectron2 for the names that are assigned to the logged  metrics/losses? We're currently using the following set of YOLOv5 labels for the data we log to TensorBoard and W&B, but there is a PR https://github.com/ultralytics/yolov5/pull/3115 open to align these more closely with MM. What do you think is the best approach here?

```python
            tags = ['train/box_loss', 'train/obj_loss', 'train/cls_loss',  # train loss
                    'metrics/precision', 'metrics/recall', 'metrics/mAP_0.5', 'metrics/mAP_0.5:0.95',  # metrics
                    'val/box_loss', 'val/obj_loss', 'val/cls_loss',  # val loss
                    'x/lr0', 'x/lr1', 'x/lr2']  # learning rates for the 3 YOLOv5 parameter groups
```

**Comment by Borda** [comment]
PL is independent on logging names but these you mentioned sounds fine to me =)

**Comment by glenn-jocher** [comment]
@Borda ok thanks!

**Comment by fcakyon** [comment]
@glenn-jocher what do you think is the best solution for this? Additional arguments for changing the logging format or breaking changes in the overall logging format?

**Comment by glenn-jocher** [comment]
@fcakyon I'm not sure. There's a lack of consensus on naming standards unfortunately, so there seems to be no clear standard to align ourselves to at the moment.

**Comment by fcakyon** [comment]
@glenn-jocher I agree. I think it is safer to take the logger type as argument, instead of making breaking changes (as in #3115). Dont you think so?

**Comment by glenn-jocher** [comment]
@fcakyon no I think there's already too many training arguments, it's probably overwhelming scanning through them for newcomers, which is not what we want. But I'm open to improving our naming conventions once. The LR improvements look good, not sure really about the other ones though.
